src/Stratis.Bitcoin.Features.RPC/Controllers/FullNodeController.cs

```diff
@@ -175,6 +177,44 @@ public BlockHeaderModel GetBlockHeader(string hash, bool isJsonFormat = true)
             return model;
         }
 
+        /// <summary>
+        /// Return information about <address>.
+        /// </summary>
+        /// <param name="address">BitcoinAddress to validate.</param>
+        /// <returns>information about <address>. {isvalid}</returns>
+        [ActionName("validateaddress")]
+        [ActionDescription("Returns information about a bitcoin address")]
+        public JObject ValidateAddress(string address)
+        {
+            if(address == null)
+                throw new ArgumentNullException("address");
+
+            JObject res = new JObject();
+            res["isvalid"] = false; // until proven otherwise
+
+            // P2PKH
+            if(BitcoinPubKeyAddress.IsValid(address, ref this.Network))
```

Comment:
With this logic you will effectively be evaluating each address type. It'll be better to return once a valid address has been found? Maybe look into refactoring this a bit 👍

Reply:
Really? I was under the impression `else if` should only execute if the prior conditional control statements evaluate to false i.e. no valid address has been found. Else the result will be returned. I chose not to return because bitcoin/bitcoin has other fields returned in res, of which we need at least one for TumbleBit (HdAddress)